src/Aspire.Hosting/Dcp/ApplicationExecutor.cs

```diff
@@ -185,25 +187,51 @@ private void WatchResourceChanges(CancellationToken cancellationToken)
 
         async Task WatchKubernetesResource<T>(Func<WatchEventType, T, Task> handler) where T : CustomResource
         {
+            var retryUntilCancelled = new RetryStrategyOptions()
+            {
+                ShouldHandle = new PredicateBuilder().HandleInner<EndOfStreamException>(),
+                BackoffType = DelayBackoffType.Exponential,
+                MaxRetryAttempts = int.MaxValue,
+                UseJitter = true,
+                MaxDelay = TimeSpan.FromSeconds(30),
```

Comment:
30 seconds seems short no?

Reply:
Discussed offline. This is the only code that is hitting the dcp resource watch endpoint, hitting it every 30 seconds is probably fine. We hit it more frequently than that with an exponential backoff to 30 seconds. Realistically - if it doesn't respond by 30 seconds it is probably dead and a different (fatal) exception would have thrown and we would have aborted.